match/lib/match/encrypt.rb

```diff
@@ -81,43 +81,45 @@ def iterate(source_path)
       end
     end
 
-    def crypt(path: nil, password: nil, encrypt: true)
-      if password.to_s.strip.length == 0 && encrypt
-        UI.user_error!("No password supplied")
-      end
+    def encrypt(path: nil, password: nil)
+      UI.user_error!("No password supplied") if password.to_s.strip.length == 0
 
-      tmpfile = File.join(Dir.mktmpdir, "temporary")
-      command = ["openssl aes-256-cbc"]
-      command << "-k #{password.shellescape}"
-      command << "-in #{path.shellescape}"
-      command << "-out #{tmpfile.shellescape}"
-      command << "-a"
-      command << "-d" unless encrypt
-      command << "&> /dev/null" unless FastlaneCore::Globals.verbose? # to show an error message if something goes wrong
-
-      _out, err, st = Open3.capture3(command.join(' '))
-      success = st.success?
-
-      # Ubuntu `openssl` does not fail on failure
-      # but at least outputs an error message
-      unless err.to_s.empty?
-        success = false
+      data_to_encrypt = ""
+      File.open(path) do |file|
+        data_to_encrypt = file.read
       end
+      salt = SecureRandom.random_bytes(8)
 
-      UI.crash!("Error decrypting '#{path}'") unless success
+      cipher = OpenSSL::Cipher.new('AES-256-CBC')
+      cipher.encrypt
+      cipher.pkcs5_keyivgen(password, salt, 1, "MD5")
+      encrypted_data = "Salted__" + salt + cipher.update(data_to_encrypt) + cipher.final
 
-      # On non-Mac systems (more specific Ubuntu Linux) it might take some time for the file to actually be there (see #11182).
-      # To try to circumvent this flakyness (in tests), we wait a bit until the file appears (max 2s) (usually only 0.1 is actually waited)
-      unless FastlaneCore::Helper.is_mac?
-        count = 0
-        # sleep until file exists or 20*0.1s (=2s) passed
-        until File.exist?(tmpfile) || count == 20
-          sleep(0.1)
-          count += 1
-        end
+      File.open(path, "w") do |file|
+        file.write(Base64.encode64(encrypted_data))
+      end
+    end
+
+    def decrypt(path: nil, password: nil)
+      stored_data = ""
+      File.open(path) do |file|
+        stored_data = Base64.decode64(file.read)
       end
+      salt = stored_data[8..15]
+      data_to_decrypt = stored_data[16..-1]
+
+      decipher = OpenSSL::Cipher.new('AES-256-CBC')
+      decipher.decrypt
+      decipher.pkcs5_keyivgen(password, salt, 1, "MD5")
 
-      FileUtils.mv(tmpfile, path)
+      decrypted_data = decipher.update(data_to_decrypt) + decipher.final
+
+      File.open(path, "w") do |file|
+        file.write(decrypted_data)
+      end
+    rescue => error
```

Comment:
* any reason we catch errors only in decrypt and not in encrypt?
* why logging the error only in verbose?